security/providers/oidc/src/main/java/io/helidon/security/providers/oidc/OidcProvider.java

```diff
@@ -165,30 +170,22 @@ private String findTenantIdFromRedirects(ProviderRequest providerRequest) {
             }
         }
         if (oidcConfig.useCookie() && tenantId.isEmpty()) {
-            tenantId = oidcConfig.tenantCookieHandler()
-                    .findCookie(providerRequest.env().headers())
-                    .stream()
-                    .map(this::getValueFromSingle)
-                    .findFirst();
+            Optional<Single<String>> cookie = oidcConfig.tenantCookieHandler()
+                    .findCookie(providerRequest.env().headers());
 
-            if (tenantId.isEmpty()) {
-                missingLocations.add("cookie");
+            if (cookie.isPresent()) {
+                return cookie.get();
             }
+            missingLocations.add("cookie");
         }
         if (tenantId.isPresent()) {
-            return tenantId.get();
+            return Single.just(tenantId.get());
         } else {
-            LOGGER.finest(() -> "Missing tenant id, could not find in either of: " + missingLocations);
-            LOGGER.finest(() -> "Falling back to the default tenant id: " + DEFAULT_TENANT_ID);
-            return DEFAULT_TENANT_ID;
-        }
-    }
-
-    private String getValueFromSingle(Single<String> cookieSingle) {
-        try {
-            return cookieSingle.get();
-        } catch (Exception e) {
-            throw new RuntimeException(e);
+            if (LOGGER.isLoggable(Level.FINEST)) {
+                LOGGER.finest(() -> "Missing tenant id, could not find in either of: " + missingLocations + "\n"
```

Comment:
when guarded by `isLoggable`, do not use a lambda

Reply:
Also do not add new lines to log output, just keep it as a single line, single message